File: DesignPatternsDotNet/ProxyPattern/StatementInformer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace ProxyPattern
{
    public class StatementInformer: IBalanceInformation
    {
        private IBalanceInformation myAccount = null;
        public StatementInformer(IBalanceInformation account)
        {
            myAccount = account;
        }
        public StatementInformer()
        {

        }
        public double GetBalance()
        {
            if (myAccount == null)
            {
                myAccount = new Account();
            }
            return myAccount.GetBalance();
        }
    }
}

File: DesignPatternsJava/src/main/java/com/proxy/StatementInformer.java
package com.proxy;

public class StatementInformer implements IBalanceInformation{
	IBalanceInformation myAcccount = null;
	public StatementInformer() {
	}
	public StatementInformer(IBalanceInformation account) {
		myAcccount=account;
	}
	public double getBalance() {
		if(myAcccount==null){
			myAcccount = new Account();
		}		
		return myAcccount.getBalance();
	}
	public String lastAccesDate()
	{
		Account aRealaccount = new Account();
		return aRealaccount.getLastAccessDate(); 
	}
}
